plugins/vault-backend/src/service/vaultApi.ts

```diff
@@ -59,11 +59,17 @@ export interface VaultApi {
    * Returns the URL to access the Vault UI with the defined config.
    */
   getFrontendSecretsUrl(): string;
+
   /**
    * Returns a list of secrets used to show in a table.
    * @param secretPath - The path where the secrets are stored in Vault
+   * @param secretMount - The mount point of the secrets engine, optional, overrides default secret engine
    */
-  listSecrets(secretPath: string): Promise<VaultSecret[]>;
+  listSecrets(
+    secretPath: string,
+    secretMount?: string | undefined,
```

Comment:
I'll also add that when you add a second optional argument like this, you get limited in your ability to iterate on the interface in the future without making it a breaking change. We often make an options object instead, so that the arguments get names in the code when you call it, and you can easily add more fields without breakage.

```ts
listSecrets(
  secretPath: string,
  options?: {
    secretEngine?: string;
  },
)
```

same below